src/ol/featureloader.js

```diff
@@ -86,35 +86,39 @@ export function loadFeaturesXhr(
     // status will be 0 for file:// urls
     if (!xhr.status || (xhr.status >= 200 && xhr.status < 300)) {
       const type = format.getType();
-      /** @type {Document|Node|Object|string|undefined} */
-      let source;
-      if (type == 'json') {
-        source = JSON.parse(xhr.responseText);
-      } else if (type == 'text') {
-        source = xhr.responseText;
-      } else if (type == 'xml') {
-        source = xhr.responseXML;
-        if (!source) {
-          source = new DOMParser().parseFromString(
-            xhr.responseText,
-            'application/xml',
+      try {
+        /** @type {Document|Node|Object|string|undefined} */
+        let source;
+        if (type == 'json') {
+          source = JSON.parse(xhr.responseText);
+        } else if (type == 'text') {
+          source = xhr.responseText;
+        } else if (type == 'xml') {
+          source = xhr.responseXML;
+          if (!source) {
+            source = new DOMParser().parseFromString(
+              xhr.responseText,
+              'application/xml',
+            );
+          }
+        } else if (type == 'arraybuffer') {
+          source = /** @type {ArrayBuffer} */ (xhr.response);
+        }
+        if (source) {
+          success(
+            /** @type {Array<FeatureType>} */
+            (
+              format.readFeatures(source, {
+                extent: extent,
+                featureProjection: projection,
+              })
+            ),
+            format.readProjection(source),
           );
+        } else {
```

Comment:
Striclty speaking this `if-else` is not needed anymore, since a falsy source would throw an error which is then caught and calls `failure();`, but it seemed cleaner to keep it.